src/ui/public/visualize/visualize.js

```diff
@@ -121,14 +121,17 @@ uiModules
           });
         }, 100);
 
-        $scope.vis.on('update', () => {
+        //todo: clean this one up as well
```

Comment:
What's the reason not to implement that directly?

Reply:
i am wondering the same ?